packages/istanbul-lib-report/lib/tree.js

```diff
@@ -59,97 +62,80 @@ util.inherits(CompositeVisitor, Visitor);
     };
 });
 
-function Node() {}
-
-/* istanbul ignore next: abstract method */
-Node.prototype.getQualifiedName = function() {
-    throw new Error('getQualifiedName must be overridden');
-};
-
-/* istanbul ignore next: abstract method */
-Node.prototype.getRelativeName = function() {
-    throw new Error('getRelativeName must be overridden');
-};
-
-/* istanbul ignore next: abstract method */
-Node.prototype.isRoot = function() {
-    return !this.getParent();
-};
-
-/* istanbul ignore next: abstract method */
-Node.prototype.getParent = function() {
-    throw new Error('getParent must be overridden');
-};
-
-/* istanbul ignore next: abstract method */
-Node.prototype.getChildren = function() {
-    throw new Error('getChildren must be overridden');
-};
+class Node {
+    /* istanbul ignore next: abstract method */
```

Comment:
is this really an abstract method?

Reply:
Currently it is.  A follow-up patch will actually make a constructor for `class Node` that will accept a `root` argument so `getRoot` will no longer be abstract.

Reply:
Interesting... it has a defined body though, so isn't it not-abstract by definition?

Reply:
If you mean abstract like C++ classes have abstract methods, that's not a feature of JS.  In any case these "abstract" functions are replaced in all cases so it could probably be deleted since these are internal classes that are overridden and the "abstract" functions are implemented in child classes.

Reply:
Yeah sorry it's not "strictly" abstract - anyway, I agree, I think it should just be deleted... but maybe as a follow up.